### PR TITLE
feat: add preview provider for emf files

### PR DESCRIPTION
**Pull request by kesselb**
* Target version: main

### Summary

For https://github.com/nextcloud/files_emfviewer/issues/1
In addition to https://github.com/nextcloud/server/pull/41395

We added support for image/emf via LibreOffice for 28. 

For setups already using Richdocuments/Collabora it's nicer to let the previews generate via collabora web and no local LibreOffice installation is required. 

### TODO

- [x] CI
- [ ] Test
- [ ] Documentation

### Checklist

- [ ] Code is properly formatted
- [ ] Sign-off message is added to all commits
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -88,6 +89,7 @@ public function register(IRegistrationContext $context): void {
 			'getPathForToken',
 			'getWopiForToken',
 		]);
+		$context->registerPreviewProvider(EMF::class, '/image\/emf/');
```
Comment (juliushaertl) [inline lib/AppInfo/Application.php:92]: Did you give that a test with a Collabora intance setup? We have not migrated our preview class to the IProviderV2 interface (which we really should 🙈 ) and as far as I see it the context registration requires that.

Comment (kesselb) [reply lib/AppInfo/Application.php]: > Did you give that a test with a Collabora intance setup?

Yep, it works like a charm ;)

> the context registration requires that.

True, the psalm-param type hints IProviderV2.

I guess it's better to use the deprecated way that expects a IProvider instance for now.

Comment (juliushaertl) [reply lib/AppInfo/Application.php]: Yep, let's stick to that then for now

**Review by juliushaertl — APPROVED** [review]

**Comment by kesselb** [comment]
Documentation: https://github.com/nextcloud/documentation/pull/11292

**Pull request merged**